From c556ee5d92698d0d8423173376bae8a174316b3f Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Fri, 20 Jan 2023 11:02:07 +0200
Subject: [PATCH] compulab: iot-gate-imx8plus: add support for break-out add-on
 board

This patch provides break-out add-on board (aka IE-BRKOUT) usage
example by configuring following signals and interfaces:
 - PWM3 (P5-09)
 - PWM4 (P5-07)
 - GPIO5_10 (P5-13)
 - GPIO5_11 (P5-11)
 - GPIO5_12 (P5-14)
 - GPIO5_13 (P5-12)
 - I2C5: SDA (P5-08), SCL (P5-10)

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile         |  1 +
 .../iot-gate-imx8plus-brkout_pwm_gpio.dts     | 86 +++++++++++++++++++
 2 files changed, 87 insertions(+)
 create mode 100644 arch/arm64/boot/dts/compulab/iot-gate-imx8plus-brkout_pwm_gpio.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 60205dd04b1f..ccbcf186d0f7 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -21,6 +21,7 @@ dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-headless.dtb
 dtb-$(CONFIG_ARCH_MXC) += iot-gate-imx8plus.dtb
 dtb-$(CONFIG_ARCH_MXC) += iot-gate-imx8plus-usbdev.dtb
 dtb-$(CONFIG_ARCH_MXC) += iot-gate-imx8plus-m2tpm.dtb
+dtb-$(CONFIG_ARCH_MXC) += iot-gate-imx8plus-brkout_pwm_gpio.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus_mipi-csi1.dtb
diff --git a/arch/arm64/boot/dts/compulab/iot-gate-imx8plus-brkout_pwm_gpio.dts b/arch/arm64/boot/dts/compulab/iot-gate-imx8plus-brkout_pwm_gpio.dts
new file mode 100644
index 000000000000..b7cb14a84aff
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/iot-gate-imx8plus-brkout_pwm_gpio.dts
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2023 CompuLab Ltd
+ */
+
+#include "iot-gate-imx8plus.dts"
+
+/*
+ * IOT-GATE-IMX8PLUS with BRKOUT extension module in M.2 Expansion Connector
+ * Example pionout:
+ * ----+----+----------
+ *  P1 | P5 | Function
+ * ----+----+----------
+ *  11 | 09 | PWM3_OUT
+ *  09 | 07 | PWM4_OUT
+ *  21 | 13 | GPIO5_IO10
+ *  23 | 11 | GPIO5_IO11
+ *  17 | 14 | GPIO5_IO12
+ *  15 | 12 | GPIO5_IO13
+ *  34 | 08 | I2C5_SDA
+ *  36 | 10 | I2C5_SCL
+ */
+
+&gpio5 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio5>;
+	status = "okay";
+};
+
+&ecspi2 {
+	status = "disabled";
+};
+
+&i2c5 {
+	status = "okay";
+
+	eeprom_m2: eeprom-i2c5@54 {
+		compatible = "atmel,24c08";
+		pagesize = <16>;
+		reg = <0x54>;
+		status = "okay";
+	};
+};
+
+&i2c6 {
+	status = "disabled";
+};
+
+&pwm3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm3>;
+	status = "okay";
+};
+
+&pwm4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm4>;
+	status = "okay";
+};
+
+&usdhc2 {
+	status = "disabled";
+};
+
+&iomuxc {
+	pinctrl_gpio5: gpio5grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_ECSPI2_SCLK__GPIO5_IO10			0x100 /* P5-13 */
+			MX8MP_IOMUXC_ECSPI2_MOSI__GPIO5_IO11			0x100 /* P5-11 */
+			MX8MP_IOMUXC_ECSPI2_MISO__GPIO5_IO12			0x100 /* P5-14 */
+			MX8MP_IOMUXC_ECSPI2_SS0__GPIO5_IO13			0x100 /* P5-12 */
+		>;
+	};
+
+	pinctrl_pwm3: pwm3grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI5_RXC__PWM3_OUT				0x116 /* P5-09 */
+		>;
+	};
+
+	pinctrl_pwm4: pwm4grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI5_RXFS__PWM4_OUT			0x116 /* P5-11 */
+		>;
+	};
+};
-- 
2.20.1

